iot-edge/Tutorials/PythonFilter/main.py

```diff
@@ -5,56 +5,66 @@
 # For guidance, see https://docs.microsoft.com/azure/iot-edge/tutorial-python-module
 
 import sys
-import time
 import threading
-from azure.iot.device import IoTHubModuleClient, Message
+import signal
+from azure.iot.device import IoTHubModuleClient
 
 # global counters
 RECEIVED_MESSAGES = 0
 
-def receive_message_listener(client):
-    # This listener function only triggers for messages sent to "input1".
-    # Messages sent to other inputs or to the default will be silently discarded.
-    global RECEIVED_MESSAGES
-    while True:
-        message = client.receive_message_on_input("input1")   # blocking call
-        RECEIVED_MESSAGES += 1
-        print("Message received on input1")
-        print( "    Data: <<{}>>".format(message.data) )
-        print( "    Properties: {}".format(message.custom_properties))
-        print( "    Total calls received: {}".format(RECEIVED_MESSAGES))
-        print("Forwarding message to output1")
-        client.send_message_to_output(message, "output1")
-        print("Message successfully forwarded")
+def create_client():
+    client = IoTHubModuleClient.create_from_edge_environment()
 
-def main():
-    try:
-        print ( "\nPython {}\n".format(sys.version) )
-        print ( "IoT Hub Client for Python" )
+    # define function for handling received messages
+    def receive_message_handler(message):
+        # NOTE: This function only handles messages sent to "input1".
+        # Messages sent to other inputs or to the default will be discarded.
+        global RECEIVED_MESSAGES
+        if message.input_name == "input1":
+            RECEIVED_MESSAGES += 1
+            print("Message received on input1")
+            print( "    Data: <<{}>>".format(message.data) )
+            print( "    Properties: {}".format(message.custom_properties))
+            print( "    Total calls received: {}".format(RECEIVED_MESSAGES))
+            print("Forwarding message to output1")
+            client.send_message_to_output(message, "output1")
+            print("Message successfully forwarded")
+        else:
+            print("Message received on unknown input: {}".format(message.input_name))
 
-        client = IoTHubModuleClient.create_from_edge_environment()
+    # Set handler
+    client.on_message_received = receive_message_handler
```

Comment:
Again, here an error could get raised. While the need to shutdown isn't strictly necessary if only attaching one handler, given that this is happening in Edge it miiiight be best to wrap this section, or alternatively, this whole function in a try/except as well so that any error would get reported via print statement.

On the other hand, it's kind of overkill and might make the sample less readable. Thoughts?